pkg/segment/writer/rawchecker.go

```diff
@@ -52,10 +52,18 @@ func ApplySearchToMatchFilterRawCsg(match *MatchFilter, col []byte) (bool, error
 	if match.MatchOperator == And {
 		var foundQword bool = true
 		if match.MatchType == MATCH_PHRASE {
-			foundQword = utils.IsSubWordPresent(col[idx:idx+clen], match.MatchPhrase)
+			regexp := match.Regexp
+			if regexp == nil {
+				return false, errors.New("Failed to compile regular expression for match phrase")
```

Comment:
this will not be an error based on the vey top comment above, we will do regexp search only if it contains *